### PR TITLE
Add wsj0mix dataset

### PR DESCRIPTION
**Pull request by mthrok**
#897

 - Add wsj0mix dataset
 - Add test to `test/torchaudio_unittest`

### CODE REVIEW

File: examples/source_separation/utils/dataset/wsj0mix.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
+from typing import Union
+from pathlib import Path
+from collections import namedtuple
+
+import torch
+from torch.utils.data import Dataset
+
+import torchaudio
+
+Sample = namedtuple("Sample", ["sample_rate", "mix", "src"])
```
Comment (vincentqb) [inline examples/source_separation/utils/dataset/wsj0mix.py:10]: [same comment](https://github.com/pytorch/audio/pull/882#discussion_r474088882)

**Comment by vincentqb** [comment]
as mentioned [here](https://github.com/pytorch/audio/issues/897#issuecomment-683139313), training on an open dataset like LibriMix would allow for more users to experiment with the dataset

File: examples/source_separation/utils/dataset/wsj0mix.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        self.files.sort()
+
+    def _load_audio(self, path) -> torch.Tensor:
+        waveform, sample_rate = torchaudio.load(path)
+        if sample_rate != self.sample_rate:
+            raise ValueError(
+                f"The dataset contains audio file of sample rate {sample_rate}. "
+                "Where the requested sample rate is {self.sample_rate}."
+            )
+        return waveform
```
Comment (vincentqb) [inline examples/source_separation/utils/dataset/wsj0mix.py:46]: What does this function serve beyond wrapping load?

**Review by cpuhrsch — APPROVED** [review]

File: examples/source_separation/utils/dataset/utils.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
+from typing import List
+from functools import partial
+from collections import namedtuple
+
+import torch
+
+from . import wsj0mix
+
+Batch = namedtuple("Batch", ["mix", "src", "mask"])
```
Comment (vincentqb) [inline examples/source_separation/utils/dataset/utils.py:9]: name may be misleading?

**Review by vincentqb — APPROVED** [review]
Minor comments, but LGTM

File: examples/source_separation/utils/dataset/wsj0mix.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        self,
+        root: Union[str, Path],
+        num_speakers: int,
+        sample_rate: int,
+        audio_ext: str = ".wav",
+    ):
+        self.root = Path(root)
+        self.sample_rate = sample_rate
+        self.mix_dir = (self.root / "mix").resolve()
+        self.src_dirs = [(self.root / f"s{i+1}").resolve() for i in range(num_speakers)]
```
Comment (cpuhrsch) [inline examples/source_separation/utils/dataset/wsj0mix.py:34]: nit: could use os path join

**Comment by mthrok** [comment]
thanks!

**Pull request merged**